### PR TITLE
nats-streaming-server 0.2.2 (new formula)

### PR DESCRIPTION
**Pull request by ColinSullivan1**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/nats-streaming-server.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class NatsStreamingServer < Formula
+  desc "Lightweight cloud messaging system"
+  homepage "https://nats.io"
+  url "https://github.com/nats-io/nats-streaming-server/archive/v0.2.2.tar.gz"
+  sha256 "741d03db1f78c348856476b0b27f39c391a6695aa6c199997a517e6f8c0e58f1"
+  head "https://github.com/nats-io/nats-streaming-server.git"
+
+  bottle do
```
Comment (DomT4) [inline Formula/nats-streaming-server.rb]: Please don't add a bottle block to new formulae. Our CI will take care of it.

File: Formula/nats-streaming-server.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    sha256 "a5f81cb6646f4c7d19d035cb7f8b4de02c946200702ec38983f677ea6e89a73f" => :mavericks
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p "src/github.com/nats-io"
+    ln_s buildpath, "src/github.com/nats-io/nats-streaming-server"
+    @buildfile = buildpath/"src/github.com/nats-io/nats-streaming-server/nats-streaming-server.go"
```
Comment (DomT4) [inline Formula/nats-streaming-server.rb]: Don't need the `@`; you're only calling it from inside the same block.

File: Formula/nats-streaming-server.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    system "go", "build", "-v", "-o", buildpath/bin/"nats-streaming-server", @buildfile
```
Comment (DomT4) [inline Formula/nats-streaming-server.rb]: `buildpath/bin/"nats-streaming-server"` can probably just be `bin/"nats-streaming-server"`?

File: Formula/nats-streaming-server.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  test do
+    pid = fork do
+      exec "#{bin}/nats-streaming-server --port=8085 --pid=#{testpath}/pid --log=#{testpath}/log"
+    end
+    sleep 3
+
+    begin
+      assert_match "INFO", shell_output("curl localhost:8085")
+      assert File.exist?(testpath/"log")
+      assert_match version.to_s, File.open(testpath/"log").grep(/Starting nats-streaming-server/).first
```
Comment (DomT4) [inline Formula/nats-streaming-server.rb]: `File.read` might be a simpler choice than the logic here.

**Comment by ColinSullivan1** [comment]
@DomT4 - Thanks for the review!  I'll update the PR.

**Comment by MikeMcQuaid** [comment]
👍 from me if @DomT4 is happy.

**Comment by DomT4** [comment]
Seems alright.

Merged in https://github.com/Homebrew/homebrew-core/commit/886f894d6e63d5a850afc001450e1f6b788b1786. Thank you for another contribution to Homebrew @ColinSullivan1; we appreciate it! :smiley_cat:

**Pull request closed without merging**